src/util/colorScale.js

```diff
@@ -120,10 +120,20 @@ export const calcColorScale = (colorBy, controls, tree, treeToo, metadata) => {
     console.error("calcColorScale called before tree is ready.");
     error = true;
   } else if (genotype) { /* G E N O T Y P E */
-    legendValues = orderOfGenotypeAppearance(tree.nodes);
+    legendValues = orderOfGenotypeAppearance(tree.nodes, controls.mutType);
+    const trueValues = controls.mutType === "nuc" ? legendValues.filter((x) => x !== "X" && x !== "-" && x !== "N") :
+      legendValues.filter((x) => x !== "X" && x !== "-");
+    const unkValues = controls.mutType === "nuc" ? legendValues.filter((x) => x === "X" || x === "-" || x === "N") :
+      legendValues.filter((x) => x === "X" || x === "-");
+    let domain = [undefined, ...trueValues];
+    let range = [unknownColor, ...genotypeColors.slice(0, trueValues.length)];
+    if (unkValues.length) { // we must add these to the domain + provide a value in the range
```

Comment:
An alternate way to do this would be 
```
if (legendValues.includes("X") || legendValues.includes("-") || (controls.mutType === "nuc" && legendValues.includes("N"))
```

Reply:
To add specific colours per gap, X or N, I ended up taking a different route. But it seems like there must be a more efficient way to do this...